scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala

```diff
@@ -0,0 +1,32 @@
+package scala.meta.internal.scalahost
+
+import scala.meta.semantic.v1.CompilerMessage
+import scala.reflect.internal.util.Position
+import scala.tools.nsc.reporters.Reporter
+import scala.tools.nsc.reporters.StoreReporter
+
+import org.scalameta.logger
+
+/** Reporter that forwards messages to underlying reporter AND stores messages. */
+class ScalahostReporter(underlying: Reporter) extends StoreReporter {
```

Comment:
Sure, but where do we make use of it?

upd. Aaahhh, I see now. `CompilationUnit.messages` is an extension method. I could bet that there was such method already in scalac.